openstack/utils/choose_version.go

```diff
@@ -109,3 +113,94 @@ func ChooseVersion(client *gophercloud.ProviderClient, recognized []*Version) (*
 
 	return highest, endpoint, nil
 }
+
+// GetSupportedMicroversions returns the minimum and maximum microversion that is supported by the ServiceClient Endpoint.
+func GetSupportedMicroversions(client gophercloud.ServiceClient) (string, string, error) {
+	type valueResp struct {
+		ID         string `json:"id"`
+		Status     string `json:"status"`
+		Version    string `json:"version"`
+		MinVersion string `json:"min_version"`
+	}
+
+	type response struct {
+		Version valueResp `json:"version"`
+	}
+	var resp response
+	_, err := client.Request("GET", client.Endpoint, &gophercloud.RequestOpts{
```

Comment:
When I try this, I get a segfault
```
panic: assignment to entry in nil map

goroutine 1 [running]:
github.com/gophercloud/gophercloud.(*ServiceClient).Request(0xc00010fa90, {0x6efecb, 0x3}, {0xc00001e100, 0x1a}, 0xc?)
        /home/dtantsur/gophercloud/service_client.go:150 +0x147
github.com/gophercloud/gophercloud/openstack/utils.GetSupportedMicroversions({0xc00013c000, {0xc00001e100, 0x1a}, {0x0, 0x0}, {0x6f117e, 0x9}, {0x0, 0x0}, 0xc00002ac90})
        /home/dtantsur/gophercloud/openstack/utils/choose_version.go:130 +0xd1
main.main()
        /home/dtantsur/gophercloud/main/main.go:26 +0x98
```

Reply:
https://github.com/gophercloud/gophercloud/pull/2799

Reply:
You can avoid it by using Get instead of Request for now.

Reply:
Fixed